src/main/java/com/bringframework/registry/BeanDefinitionRegistry.java

```diff
@@ -0,0 +1,36 @@
+package com.bringframework.registry;
+
+import com.bringframework.context.ApplicationContext;
+import java.util.Map;
+
+/**
+ * Interface for a registry that holds {@link BeanDefinition}.
+ *
+ * <p>This is used by the {@link ApplicationContext} to create and manage bean instances.
+ **/
+public interface BeanDefinitionRegistry {
+
+  /**
+   * Register a {@link BeanDefinition} with the given name.
+   *
+   * @param name           the name of the bean definition
+   * @param beanDefinition the {@link BeanDefinition} instance to register
+   */
+  void registerBeanDefinition(String name, BeanDefinition beanDefinition);
+
+  /**
+   * Retrieve the {@link BeanDefinition} for the given bean name.
+   *
+   * @param name the name of the bean definition
+   * @return the {@link BeanDefinition} for the given name or {@code null} if none is found
```

Comment:
Suggestion: instead of returning null : 1) return Optional 2) throw NoSuchBeanException

Reply:
I removed this part from the documentation.
I think that better should be thrown some exception
Will be done int the next tickets.

Reply:
Good! Makes sense then